packages/salesforce/src/Adaptor.js

```diff
@@ -515,90 +509,54 @@ export function insert(sObjectName, records) {
 
 /**
  * Executes an SOQL (Salesforce Object Query Language) query to retrieve records from Salesforce.
- * This operation uses {@link https://jsforce.github.io/document/#using-soql for querying salesforce records} using SOQL query and handles pagination.
  * Note that in an event of a query error, error logs will be printed but the operation will not throw the error.
  *
  * The Salesforce query API is subject to rate limits, {@link https://sforce.co/3W9zyaQ learn more here}.
- *
  * @public
  * @example <caption>Run a query and download all matching records</caption>
- * query('SELECT Id FROM Patient__c', { autoFetch: true });
+ * query('SELECT Id FROM Patient__c', { max: false });
+ * @example <caption>Run a query and limit records</caption>
+ * query('SELECT Id From Account Limit 10');
```

Comment:
Oh hey this is a great point, I hadn't considered that users can limit in the query itself.

This makes me think that we should actually default max to infinity, which would avoid a conflict on limit and max

Salesforce users used to running big queries are likely to limit as standard, and don't need our API getting in the way. They probably want control from the query, right?

Reply:
If we default `max: Infinity` we are partially doing `bulkQuery()` job. Because the only time you will need to use bulkQuery will be when you're query string exceed 4000 characters. otherwise just use `query()`.

There is also a difference between using `max: 100` to `select id from account limit 100`. 

With `max: 100` you will get 100 records + metadata for the next records. 
But  will `limit 100` in a query` you will only get `100 records`

Reply:
I like the idea of defaulting `max: Infinity` but i am concerned if user have 1million records. Do we really want them to use `query()` function ? Shouldn't we have a default limit even if it's high Eg: 30k records ??

Reply:
Hiya @josephjclark, I have replaced `max` with `limit` and updated docs, test and add warning if query contain `LIMIT` or `limit` clause. 

Regarding the default value. I think it should be 10k which matches the internal default maxFetch, I think 5k is very small number of records